src/mammos_spindynamics/uppasd/_simulation.py

```diff
@@ -0,0 +1,458 @@
+
+        if verbosity >= 1:
+            print(
+                f"Running simulations for {len(Ts)} different temperatures:\n    {Ts!s}"
+            )
+
+        # run first simulation with default options; later simulations optionally with
+        # restarting from previous
+        if verbosity >= 2:
+            print(f"T={Ts[0]}:", end=" ")
+        run_data = self.run(
+            T=Ts[0],
+            out=run_path,
+            uppasd_executable=uppasd_executable,
+            verbosity=verbosity - 1,
+            **kwargs,
+        )
+        for T_ in Ts[1:]:
+            if restart_with_previous:
+                kwargs.update({"initmag": 4, "restartfile": run_data.restartfile})
+            if verbosity >= 2:
+                print(f"T={T_}:", end=" ")
+            run_data = self.run(
+                T=T_,
+                out=run_path,
+                uppasd_executable=uppasd_executable,
+                verbosity=verbosity - 1,
+                **kwargs,
+            )
+
+        result = _data.TemperatureSweepData(run_path)
+        result.save_output(run_path)
+        return result
+
+
+def _create_run_dir(base: Path, mode="run") -> tuple[Path, int]:
+    """Create a new directory for run or temperature_sweep.
+
+    The directory name is prefixed with an index, which enumerates all runs and
+    temperature_sweeps. The function looks for the maximum existing index and uses
+    the next index.
+    """
+    if mode not in ["run", "temperature_sweep"]:
+        raise ValueError(
+            f"Mode {mode} not supported, must be 'run' or 'temperature_sweep'"
+        )
+
+    if not base.exists():
+        base.mkdir(parents=True)
+    elif base.is_file():
+        raise RuntimeError(f"The path '{base}' passed as output directory is a file.")
+
+    if not (base / "mammos_spindynamics.yaml").exists():
+        with open(base / "mammos_spindynamics.yaml", "w") as f:
+            yaml.dump({"metadata": {"mode": "mammos_uppasd_data"}}, f)
+
+    run_indices = [
+        int(p.name.split("-")[0])
+        for p in base.glob("*")
+        if re.match(r"^\d+-(run|temperature_sweep)$", p.name)
+    ]
+    next_index = max(run_indices) + 1 if run_indices else 0
+
+    next_run_path = base / f"{next_index}-{mode}"
+    # The next call would fail if the directory exists already. This should never
+    # happen. We can rely on it as additional safety-check to not overwrite anything.
+    next_run_path.mkdir()
+    return next_run_path, next_index
+
+
+def _write_inputs(
+    run_path: Path,
+    inp_file_content: str,
+    files_to_copy: dict[str, Path],
+    metadata: dict[str, Any],
+) -> None:
+    """Write input files.
+
+    The function writes the file ``inpsd.dat``, copies all files in ``files_to_copy``,
+    where keys are the new filenames and values are paths to the original files, and
+    create a ``mammos_spindynamics.yaml`` file with ``metadata``.
+
+    The directory ``out_path`` must already exist.
```

Comment:
```suggestion
    The directory ``run_path`` must already exist.
```

But maybe I prefer `out_path` for the whole function